src/core/audio/AudioDevice.cpp

```diff
@@ -45,53 +49,69 @@ AudioDevice::AudioDevice( const ch_cnt_t _channels, AudioEngine*  _audioEngine )
 
 AudioDevice::~AudioDevice()
 {
-	delete[] m_buffer;
 	m_devMutex.tryLock();
 	unlock();
 }
 
-
-
-
-void AudioDevice::processNextBuffer()
+bool AudioDevice::nextBuffer(InterleavedBufferView<float> dst)
 {
-	const fpp_t frames = getNextBuffer( m_buffer );
-	if (frames) { writeBuffer(m_buffer, frames); }
-	else
+	if (!m_running.test_and_set(std::memory_order_acquire))
```

Comment:
Are you using `m_running` like a mutex? Why wasn't something like this needed in `AudioDevice` prior to this PR?